phoenix-core/src/main/java/org/apache/phoenix/iterate/TableResultIterator.java

```diff
@@ -124,6 +125,7 @@ public TableResultIterator(MutationState mutationState, Scan scan, ScanMetricsHo
         this.scanMetricsHolder = scanMetricsHolder;
         this.plan = plan;
         PTable table = plan.getTableRef().getTable();
+        IndexUtil.setScanAttributes(scan, table, plan.getContext().getConnection());
```

Comment:
Most of the code in this constructor are simple assignments, but this one stands out with special behavior. It would be good to move it towards the end (unless it's crucial to invoke it right here) with a comment block.

Reply:
Ok, will move to the end